### PR TITLE
Cluster-autoscaler: Frequently Asked Questions

### PR DESCRIPTION
**Pull request by mwielgus**
cc: @MaciekPytel @fgrzadkowski

### CODE REVIEW

File: cluster-autoscaler/faq.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+Hopefully it is not a bug in Cluster Autoscaler but most likely a problem with the cluster.
+
+* Check If cluster autoscaler is up and running. In version 0.5 it periodically publishes the kube-system/cluster-autoscaler-status config map. Check last update time annotation. Should be no more than 3 min (usually 10 sec old).
+* Check kube-system/cluster-autoscaler-status if the cluster and node groups are in the healthy state. If not, check the unready nodes. 
+
+* If you expect some nodes to be deleted but they are not deleted for a long time check:
+    * if they contain pods that prevent the node from being deleted (see the corresponding question in the faq).
+    * the min/max boundaries you declared for a particular node group.
+    * check /var/log/cluster-autoscaler.log
```
Comment (MaciekPytel) [inline cluster-autoscaler/faq.md]: no dot at the end of bullet

File: cluster-autoscaler/faq.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    * events added by CA to the pod.
+    * events on the kube-system/cluster-autoscaler-status config map. 
+    * if you have quota on your cloud provider side. 
+    * check /var/log/cluster-autoscaler.log.
+
+## How can I check what is going on in CA ?
+
+There are three options:
+
+* Logs on the master node, in /var/log/cluster-autoscaler.log
```
Comment (MaciekPytel) [inline cluster-autoscaler/faq.md]: also no dot

File: cluster-autoscaler/faq.md (hunk 1 above)
Comment (MaciekPytel) [inline cluster-autoscaler/faq.md]: s/check/ in last point?

File: cluster-autoscaler/faq.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    * if they contain pods that prevent the node from being deleted (see the corresponding question in the faq).
+    * the min/max boundaries you declared for a particular node group.
+    * check /var/log/cluster-autoscaler.log
+
+* If you expect some nodes to be added to help some pending pods, but they are not added for a long time check if:
+    * the node groups that could potentially accommodate the pods are on their max size.
+    * events added by CA to the pod.
+    * events on the kube-system/cluster-autoscaler-status config map. 
+    * if you have quota on your cloud provider side. 
+    * check /var/log/cluster-autoscaler.log.
```
Comment (MaciekPytel) [inline cluster-autoscaler/faq.md]: Only first point in this list makes sense with "check if", other ones don't fit gramatically (they look as if it was "check" and not "check if").

File: cluster-autoscaler/faq.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+## How can I check what is going on in CA ?
+
+There are three options:
+
+* Logs on the master node, in /var/log/cluster-autoscaler.log
+* kube-system/cluster-autoscaler-status config map.
+* Events:
+    * on pods (particularly these that cannot be scheduled).
```
Comment (MaciekPytel) [inline cluster-autoscaler/faq.md]: s/these/those ? (not sure)

File: cluster-autoscaler/faq.md (hunk 1 above)
Comment (mwielgus) [inline cluster-autoscaler/faq.md]: done.

File: cluster-autoscaler/faq.md (hunk 2 above)
Comment (mwielgus) [inline cluster-autoscaler/faq.md]: done.

File: cluster-autoscaler/faq.md (hunk 3 above)
Comment (mwielgus) [inline cluster-autoscaler/faq.md]: done.

File: cluster-autoscaler/faq.md (hunk 4 above)
Comment (mwielgus) [inline cluster-autoscaler/faq.md]: done.

File: cluster-autoscaler/faq.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    * if they contain pods that prevent the node from being deleted (see the corresponding question in the faq).
+    * if min/max boundaries you declared for a particular node group allow the scale up.
+    * the content of /var/log/cluster-autoscaler.log.
+
+* If you expect some nodes to be added to help some pending pods, but they are not added for a long time check if:
+    * the node groups that could potentially accommodate the pods are on their max size.
+    * events added by CA to the pod.
+    * events on the kube-system/cluster-autoscaler-status config map. 
+    * if you have quota on your cloud provider side. 
+    * the content of /var/log/cluster-autoscaler.log.
```
Comment (MaciekPytel) [inline cluster-autoscaler/faq.md]: I still think this list of bullets is gramatically incorrect.

"are not added for a long time check if:" ... "the node groups that could potentially accommodate the pods are on their max size." <- works
"are not added for a long time check if:" ... "events added by CA to the pod." <- doesn't work
de groups that could potentially accommodate the pods are on their max size." <- works
"are not added for a long time check if:" ... "events added by CA to the pod." <- doesn't work
"are not added for a long time check if:" ... "events on the kube-system/cluster-autoscaler-status config map." <- doesn't work
"are not added for a long time check if:" ... "if you have quota on your cloud provider side. " <- doesn't work x2
...

**Comment by fgrzadkowski** [comment]
@mwielgus I added some question/suggestions to your original doc. Should I copy them here?

**Pull request merged**